starkup.sh

```diff
@@ -199,6 +214,16 @@ ensure() {
   if ! "$@"; then err "command failed: $*"; fi
 }
 
+get_asdf_version() {
+  asdf version 2>/dev/null | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+(-[^[:space:]]+)?$'
```

Comment:
```suggestion
  asdf --version 2>/dev/null | grep -Eo '[0-9]+\.[0-9]+\.[0-9]$'
```
nitpick. with this you only get what you need, without help mesage.

Reply:
Great catch!

I would keep the other part of regex though, as it helps us capture versions like `"0.15.0-31e8c93"`. Not sure we need it, but it's more accurate from `get_asdf_version` standpoint 🤔

Reply:
> but it's more accurate from get_asdf_version standpoint

wdym?

Reply:
> wdym?

What I mean is if `asdf --version` returns:
```
v0.15.0-31e8c93
```
it's just more accurate for function called `get_asdf_version` to return exactly what asdf returned, even if we might not need that later part 🤔

Reply:
afaik asdf follows semver in x.y.z only, so this is rather hipotethical scenario 🤔

Reply:
Well, that is the version my locally installed asdf returned 😄

Reply:
Hmmm, not sure, git installation was an official method pre-0.16 after all 🤔 
https://asdf-vm.com/guide/getting-started-legacy.html#_2-download-asdf

I don't have a strong say on this, just thought to have this as a matter of accuracy, but it's not critical tbh so I can remove it if you feel strongly we should 👍🏻

Reply:
I'm not strong as well, I just think we do not need this - after all 0.15.0 should not differ from 0.15.0-abcdfg that much (at least should not have breaking changes, which we care about the most). 

Whether you leave it or not it is fine for me, I just wanted to be sure about the purpose of this.